service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ProxyOptions.java

```diff
@@ -0,0 +1,67 @@
+package com.microsoft.azure.sdk.iot.service;
+
+import lombok.Getter;
+
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+
+/**
+ * The settings supported by this SDK when communicating to IoT Hub through a proxy. HTTP proxies are supported by this SDK,
+ * but only if the proxy does not require authentication.
+ */
+public class ProxyOptions
+{
+    @Getter
+    private Proxy proxy;
+
+    /**
+     * Create the proxy options for connecting to a proxy
+     * @param proxy the proxy to communicate through
+     */
+    public ProxyOptions(Proxy proxy)
+    {
+        if (proxy == null)
+        {
+            throw new IllegalArgumentException("Proxy cannot be null");
+        }
+
+        if (proxy.type() != Proxy.Type.HTTP)
+        {
+            throw new IllegalArgumentException("Service proxy options only support using HTTP proxies");
+        }
+
+        this.proxy = proxy;
+    }
+
+    /**
+     * @return the saved proxy hostname
+     */
+    public String getHostname()
```

Comment:
InetSocketAddress uses a capital N in getHostName. Not sure if that is the preferred casing to stick with and match for this method.

Reply:
Ah, good catch. We do use HostName and Hostname very interchangeably in this SDK. HostName seems more appropriate considering it is a pair of words